### ISSUE TITLE
[deploy] Display error message when Nexus2 staged repository fails to change state

### ISSUE THREAD

**Issue by aalmiray** [jreleaser/jreleaser#1604]
### Discussed in https://github.com/jreleaser/jreleaser/discussions/1599

<div type='discussions-op-text'>

<sup>Originally posted by **lsoares** February 16, 2024</sup>
Sorry for another question, but I think I'm almost there.
I'm getting an error in the final step:

```
Caused by: org.jreleaser.sdk.nexus2.Nexus2Exception: Unexpected error
        at org.jreleaser.sdk.nexus2.Nexus2.wrap(Nexus2.java:268)
        at org.jreleaser.sdk.nexus2.Nexus2.releaseStagingRepository(Nexus2.java:166)
        at org.jreleaser.sdk.nexus2.Nexus2MavenDeployer.deploy(Nexus2MavenDeployer.java:153)
        ... 132 more
Caused by: java.lang.IllegalStateException: repository comluissoares-1028 is not in expected state(s). Expected one of [RELEASED, NOT_FOUND], actual is CLOSED
        at org.jreleaser.sdk.nexus2.Nexus2.waitForState(Nexus2.java:194)
        at org.jreleaser.sdk.nexus2.Nexus2.lambda$releaseStagingRepository$7(Nexus2.java:170)
        at org.jreleaser.sdk.nexus2.Nexus2.wrap(Nexus2.java:262)
        ... 134 more
```

with this config:
```kotlin
jreleaser {
    gitRootSearch = true
    signing {
        active.set(Active.ALWAYS)
        armored.set(true)
    }
    deploy {
        maven {
            nexus2 {
                create("mavenCentral") {
                    active.set(Active.ALWAYS)
                    url.set("https://s01.oss.sonatype.org/service/local")
                    closeRepository.set(true)
                    releaseRepository.set(true)
                    stagingRepository("build/repos/snapshots")
                }
            }
        }
    }
}
```
thanks!</div>

As it turns out, there were validation errors that triggered and caused the repository to be marked as failed and thus did not transition to the expected state (`CLOSED`) before it could be released. See https://github.com/jreleaser/jreleaser/discussions/1599#discussioncomment-8562966

**Comment by aalmiray** [comment]
Received the following information from Maven Central team regarding queries for failed messages:

For NXRM / OSSRH publishing, the UI appears to use the [/service/local/staging/repository/{repositoryIdKey}/activity](https://s01.oss.sonatype.org/nexus-staging-plugin/default/docs/path__staging_repository_-repositoryIdKey-_activity.html) endpoint. This returns a list of activities (e.g. "open", "close", etc.) that have taken place. It appears that the "close" activity will have "events" related to the application of the rules. Events with the "name" of "ruleFailed" will have "properties" related to the issue, with "failureMessage" being the one most likely to be useful to an end-user.

**Comment by aalmiray** [comment]
Example data 

```xml
<list>
  <stagingActivity>
    <name>release</name>
    <started>2024-02-27T19:17:07.339Z</started>
    <startedByUserId>aalmiray</startedByUserId>
    <startedByIpAddress>51.154.186.252</startedByIpAddress>
    <stopped>2024-02-27T19:17:07.364Z</stopped>
    <events>
      <stagingActivityEvent>
        <timestamp>2024-02-27T19:17:07.341Z</timestamp>
        <name>rulesEvaluate</name>
        <severity>0</severity>
        <properties>
          <stagingProperty>
            <name>id</name>
            <value>nx-internal-ruleset</value>
          </stagingProperty>
          <stagingProperty>
            <name>rule</name>
            <value>RepositoryWritePolicy</value>
          </stagingProperty>
        </properties>
      </stagingActivityEvent>
      <stagingActivityEvent>
        <timestamp>2024-02-27T19:17:07.344Z</timestamp>
        <name>ruleEvaluate</name>
        <severity>0</severity>
        <properties>
          <stagingProperty>
            <name>typeId</name>
            <value>RepositoryWritePolicy</value>
          </stagingProperty>
        </properties>
      </stagingActivityEvent>
      <stagingActivityEvent>
        <timestamp>2024-02-27T19:17:07.351Z</timestamp>
        <name>ruleFailed</name>
        <severity>1</severity>
        <properties>
          <stagingProperty>
            <name>typeId</name>
            <value>RepositoryWritePolicy</value>
          </stagingProperty>
          <stagingProperty>
            <name>failureMessage</name>
            <value>Artifact updating: Repository =&apos;nosync:Does NOT Sync to Central&apos; does not allow updating artifact=&apos;/test/jreleaser/helloworld/1.0.0/helloworld-1.0.0-javadoc.jar&apos;</value>
          </stagingProperty>
          <stagingProperty>
            <name>failureMessage</name>
            <value>Artifact updating: Repository =&apos;nosync:Does NOT Sync to Central&apos; does not allow updating artifact=&apos;/test/jreleaser/helloworld/1.0.0/helloworld-1.0.0-sources.jar&apos;</value>
          </stagingProperty>
          <stagingProperty>
            <name>failureMessage</name>
            <value>Artifact updating: Repository =&apos;nosync:Does NOT Sync to Central&apos; does not allow updating artifact=&apos;/test/jreleaser/helloworld/1.0.0/helloworld-1.0.0.jar&apos;</value>
          </stagingProperty>
          <stagingProperty>
            <name>failureMessage</name>
            <value>Artifact updating: Repository =&apos;nosync:Does NOT Sync to Central&apos; does not allow updating artifact=&apos;/test/jreleaser/helloworld/1.0.0/helloworld-1.0.0.pom&apos;</value>
          </stagingProperty>
        </properties>
      </stagingActivityEvent>
      <stagingActivityEvent>
        <timestamp>2024-02-27T19:17:07.357Z</timestamp>
        <name>rulesFailed</name>
        <severity>1</severity>
        <properties>
          <stagingProperty>
            <name>id</name>
            <value>nx-internal-ruleset</value>
          </stagingProperty>
          <stagingProperty>
            <name>failureCount</name>
            <value>1</value>
          </stagingProperty>
        </properties>
      </stagingActivityEvent>
      <stagingActivityEvent>
        <timestamp>2024-02-27T19:17:07.362Z</timestamp>
        <name>repositoryReleaseFailed</name>
        <severity>1</severity>
        <properties>
          <stagingProperty>
            <name>id</name>
            <value>testjreleaser-1039</value>
          </stagingProperty>
          <stagingProperty>
            <name>cause</name>
            <value>com.sonatype.nexus.staging.StagingRulesFailedException: One or more rules have failed</value>
          </stagingProperty>
        </properties>
      </stagingActivityEvent>
    </events>
  </stagingActivity>
</list>
```

**Comment by aalmiray** [comment]
🎉 This issue has been resolved in `v1.11.0` ([Release Notes](https://github.com/jreleaser/jreleaser/releases/tag/v1.11.0))
